### PR TITLE
Video headline AB test

### PR DESCRIPTION
**Pull request by mr-mr**
## What does this change?
This AB test adds a new styled play button and promotes the video headline over the poster image.

## What is the value of this and can you measure success?
Increasing the prominence and meta info increases value of clicks and views.
Success is measured on no. of views and no. of completions vs control.

## Screenshots
mobile (with slightly increased poster image)
image

image

### CODE REVIEW

File: static/src/javascripts/bootstraps/enhanced/media/main.js [hunk 1]
```diff
@@ -146,6 +146,33 @@ define([
...
 
+    function initVideoHeadline(el, videoInfo) {
+        var buttonElement = el.parentElement.querySelector('button.vjs-big-play-button');
+        var hideControlBar = el.parentElement.querySelector('.vjs-control-bar');
+        // hideControlBar.classList.remove('vjs-control-bar');
+        hideControlBar.classList.add('vjs-control-bar--hidden');
+        buttonElement.classList.remove('vjs-big-play-button');
+        buttonElement.classList.add('vjs-big-play-button__duration');
+        var buttonDuration = el.getAttribute('data-formatted-duration');
+        buttonElement.dataset.formattedDuration = buttonDuration;
```
Comment (akash1810) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: `dataset` won't work on [IE < 11](http://caniuse.com/#search=dataset). Maybe consider using bonzo or `setAttribute`? Or target modern browsers in the test?

File: static/src/javascripts/bootstraps/enhanced/media/main.js [hunk 2]
```diff
@@ -146,6 +146,33 @@ define([
...
+        if (window.matchMedia('(max-width: 375px)').matches) {
+          mainVideoElement.classList.remove('gu-media--video');
+        }
+
+        var div = document.createElement('div');
+        var vidGradient = document.createElement('div');
+        var videoHeadline = videoInfo.title.replace(' – video','').replace(' - video','');
+        div.classList.add('video-headline');
+        vidGradient.classList.add('video-headline--gradient');
+        div.dataset.dataVideoName = videoHeadline;
```
Comment (akash1810) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: same as above

File: static/src/javascripts/bootstraps/enhanced/media/main.js [hunk 3]
```diff
@@ -146,6 +146,33 @@ define([
...
+        var hideControlBar = el.parentElement.querySelector('.vjs-control-bar');
+        // hideControlBar.classList.remove('vjs-control-bar');
+        hideControlBar.classList.add('vjs-control-bar--hidden');
+        buttonElement.classList.remove('vjs-big-play-button');
+        buttonElement.classList.add('vjs-big-play-button__duration');
+        var buttonDuration = el.getAttribute('data-formatted-duration');
+        buttonElement.dataset.formattedDuration = buttonDuration;
+        var videoPosterElement = el.parentElement.querySelector('.vjs-big-play-button__duration');
+        var mainVideoElement = videoPosterElement.parentElement;
+        mainVideoElement.querySelector('.vjs.gu-media--video');
```
Comment (akash1810) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: this is just selecting an element and isn't doing anything, can be removed

File: static/src/javascripts/bootstraps/enhanced/media/main.js (hunk 2 above)
Comment (mr-mr) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: Which line are you referring to?

Comment (akash1810) [inline static/src/javascripts/bootstraps/enhanced/media/main.js]: the use of `dataset` here - `div.dataset.dataVideoName = videoHeadline;`

File: static/src/javascripts/projects/common/modules/experiments/tests/video-headline.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        this.showForSensitive = true;
+        this.audience = 0.10;
+        this.audienceOffset = 0;
+        this.successMeasure = 'No significant difference in clicks between the variant and control';
+        this.audienceCriteria = 'All users';
+        this.dataLinkNames = '';
+        this.idealOutcome = 'Video plays in the control group is not more than 2% higher';
+
+        this.canRun = function () {
+          return document.getElementsByClassName('gu-media-wrapper--video').length > 0;
```
Comment (akash1810) [inline static/src/javascripts/projects/common/modules/experiments/tests/video-headline.js]: does this touch the carousel and fronts?

Comment (mr-mr) [inline static/src/javascripts/projects/common/modules/experiments/tests/video-headline.js]: I believe we set it to only run on articles but will double check

Comment (mr-mr) [inline static/src/javascripts/projects/common/modules/experiments/tests/video-headline.js]: @akash1810 It doesn't look like it affects either the carousel or the fronts

**Review by akash1810 — APPROVED** [review]

**Pull request merged**